internal/server/proto/server.proto

```diff
@@ -344,6 +343,22 @@ service Waypoint {
   // UpsertStatusReport updates or inserts a statusreport.
   rpc UpsertStatusReport(UpsertStatusReportRequest) returns (UpsertStatusReportResponse);
 
+  //----------------------------------------------------------------------
+  // Trigger URL Config functions
+  //----------------------------------------------------------------------
+
+  // UpsertTrigger updates or inserts a trigger URL configuration.
+  rpc UpsertTrigger(UpsertTriggerRequest) returns (UpsertTriggerResponse);
+
+  // GetTrigger returns a requested trigger message. Or an error if it does not exist.
+  rpc GetTrigger(GetTriggerRequest) returns (GetTriggerResponse);
+
+  // DeleteTrigger takes a trigger id and deletes it, if it exists.
+  rpc DeleteTrigger(DeleteTriggerRequest) returns (google.protobuf.Empty);
+
+  // ListTriggers takes a request filter, and returns any matching existing triggers
+  rpc ListTriggers(ListTriggerRequest) returns (ListTriggerResponse);
```

Comment:
I recommend in the future when you support triggering the triggers to make an RPC endpoint `RunTrigger` or something and that is what the HTTP side calls too. But, it may be worth doing the RPC alone for simplicity sake and then adding HTTP on top of that.

Reply:
That's a great idea, I'll plan to tackle that next!